repos/system_upgrade/el7toel8/actors/selinux/selinuxapplycustom/actor.py

```diff
@@ -0,0 +1,115 @@
+import os
+from shutil import rmtree
+
+from leapp.actors import Actor
+from leapp.models import SELinuxModules, SELinuxCustom, SELinuxRequestRPMs, CheckResult
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+from leapp.libraries.stdlib import run, CalledProcessError
+
+WORKING_DIRECTORY = '/tmp/selinux/'
+
+class SELinuxApplyCustom(Actor):
+    '''
+    Re-apply SELinux customizations from RHEL-7 installation
+
+    Re-apply SELinux policy customizations (custom policy modules and changes
+    introduced by semanage). Any changes (due to incompatiblity with RHEL-8
+    SELinux policy) are reported to user.
+    '''
+    name = 'selinuxapplycustom'
+    consumes = (SELinuxCustom, SELinuxModules, )
+    produces = (CheckResult, )
+    tags = (ApplicationsPhaseTag, IPUWorkflowTag, )
+
+    def process(self):
+        # cil module files need to be extracted to disk in order to be installed
+        try:
+            # clear working directory
+            rmtree(WORKING_DIRECTORY)
+        except OSError:
+            # expected
+            pass
+        try:
+            os.mkdir(WORKING_DIRECTORY)
+        except OSError:
+            self.log.info("Failed to access working directory! Aborting.")
+            return
+
+        # import custom SElinux modules
+        for semodules in self.consume(SELinuxModules):
+            self.log.info("Processing custom SELinux policy modules. Count: %d." % len(semodules.modules))
```

Comment:
Looks like log message will stay the same throughout the cycle run. Move outside to L38?

Reply:
The "for" loop is in place only in case there are multiple SELinuxModules objects (at this point there should only be one, but I was told this is the proper way to process the output of "consume").

Reply:
yeah, @vmojzis is correct about this here. Just I would probably rephrase the log message:
```suggestion
            self.log.info("Processing custom SELinux policy modules message with %d modules." % len(semodules.modules))
```